snap/snapcraft.yaml

```diff
@@ -277,26 +271,40 @@ parts:
       - libattr1-dev
       - libfdt-dev
       - curl
+      - libcapstone-dev
     build-snaps:
       - yq
     override-build: |
-      chmod +x ${SNAPCRAFT_STAGE}/qemu/scripts/configure-hypervisor.sh
-      # static build
-      echo "$(${SNAPCRAFT_STAGE}/qemu/scripts/configure-hypervisor.sh -s qemu) \
-        --disable-rbd
-        --prefix=/snap/${SNAPCRAFT_PROJECT_NAME}/current/usr \
-        --datadir=/snap/${SNAPCRAFT_PROJECT_NAME}/current/usr/share \
-        --libexecdir=/snap/${SNAPCRAFT_PROJECT_NAME}/current/usr/libexec/qemu" \
-        | xargs ./configure
+      pkg_name="qemu"
 
-      git clone https://github.com/qemu/keycodemapdb ui/keycodemapdb/
-      cd ui/keycodemapdb; git checkout 10739aa; cd ../..
+      versions_url=https://raw.githubusercontent.com/kata-containers/runtime/${SNAPCRAFT_PROJECT_VERSION}/versions.yaml
+      # arch-specific definition
+      case "$(arch)" in
+        "aarch64")
+          branch="$(curl -sSL ${versions_url} | yq r - assets.hypervisor.qemu.architecture.aarch64.branch)"
+          url="$(curl -sSL ${versions_url} | yq r - assets.hypervisor.qemu.url)"
+          commit="$(curl -sSL ${versions_url} | yq r - assets.hypervisor.qemu.architecture.aarch64.commit)"
+          patch_dir="${SNAPCRAFT_STAGE}/qemu-aarch64/patches/"
+        ;;
 
-      git clone https://github.com/qemu/capstone capstone
-      cd capstone; git checkout 22ead3e; cd ..
+        *)
+          branch="$(curl -sSL ${versions_url} | yq r - assets.hypervisor.qemu-lite.branch)"
+          url="$(curl -sSL ${versions_url} | yq r - assets.hypervisor.qemu-lite.url)"
+          commit="$(curl -sSL ${versions_url} | yq r - assets.hypervisor.qemu-lite.commit)"
+          patch_dir="${SNAPCRAFT_STAGE}/qemu-lite/patches/"
+        ;;
+      esac
+
+      # download source
+      git clone --branch ${branch} --single-branch ${url}
```

Comment:
please add `${pkg_name}` at eol

Reply:
Q. why don't clone and build it in the current directory (`git clone --branch ${branch} --single-branch ${url} .`) ?